### PR TITLE
Instagram Import

### PR DESCRIPTION
**Pull request by dansup**

### CODE REVIEW

File: app/Http/Controllers/Import/Instagram.php [hunk 1]
```diff
@@ -21,6 +22,13 @@ public function instagram()
 
     public function instagramStart(Request $request)
     {	
+        $completed = ImportJob::whereProfileId(Auth::user()->profile->id)
+            ->whereService('instagram')
+            ->whereNotNull('completed_at')
+            ->exists();
+        if($completed == true) {
+            return redirect(route('settings'))->with(['errors' => ['You can only import from Instagram once.']]);
```
Comment (Codercise) [inline app/Http/Controllers/Import/Instagram.php:30]: Is the plan here that once the user has completed their Instagram import they should use Pixelfed forever?

Comment (trwnh) [reply app/Http/Controllers/Import/Instagram.php:30]: hmm. yeah, idk about this, it should be possible to import/export multiple times ideally. that would require duplicate checking and partial archives, but it's much better than saying that you can only import once.

Comment (dansup) [reply app/Http/Controllers/Import/Instagram.php:30]: @Haydos585 That is a good point, I will investigate a solution.

Comment (Codercise) [reply app/Http/Controllers/Import/Instagram.php:30]: You could try and limit it to once a month or something. I think with a service like Pixelfed if people are going to use it they will come, add some images, leave for 6 months, try it again, leave for 3 months, try it again, stick with it.

Having the same images imported multiple times is a problem too.

Perhaps something like `from_instagram` and `instagram_id` could be added to the Post table. Code can then figure out if something has been imported before.

**Comment by dansup** [comment]
There are a few issues with the current implementation, so I will merge it for now without the routes so it can't be used until we can solve the issues @Haydos585 and @trwnh pointed out.

**Comment by bjoernfan** [comment]
Hi. I set up my own Pixelfed instance yesterday and I'm very interested in using it on a regular basis. Say I, as a non-typical user in this scenario, am happy with a one-off import - would this existing code (+ routes) serve this need today?

**Comment by aschrijver** [comment]
I'm a user in need of continuous updates. Our [Humane Tech Community](https://community.humanetech.com) has both PF and IG channels. I as fediverse fan wanna do more with PF [`@humanetech`](https://pixelfed.social/humanetech), while other members manage our [IG account](https://www.instagram.com/humane_tech_community/) (I'm not using that shit, but our members do their contribs there).

**Comment by baarkerlounger** [comment]
Same question as bjoernfan - is there a way to use this for a one-off import on my hosted instance currently?

**Comment by Rik44444** [comment]
Hi,
I wanted to ask, is this still on the agenda? Continuous sync with Insta would be amazing; single imports from insta would also already be a big pre. 
I found this, a closed Github item for this: https://github.com/pixelfed/pixelfed/issues/765 
and this item, which appears to be still active (https://github.com/pixelfed/pixelfed/pull/1046)
and this post, which appears to suggest it is ready to be launched: https://mastodon.social/@pixelfed/100087667113289786 , but i cannot find it in my instance (pixelfed.de)

**Pull request merged**